src/session/content/message_row/reply.rs

```diff
@@ -0,0 +1,288 @@
+            );
+            if show_sender {
+                let sender_name_expression = message.sender_name_expression();
+                let sender_binding =
+                    sender_name_expression.bind(&*imp.sender_label, "label", glib::Object::NONE);
+
+                bindings.push(sender_binding);
+
+                if imp.is_outgoing.get() {
+                    imp.separator.add_css_class("outgoing");
+                } else {
+                    // Color sender label
+                    if let MessageSender::User(user) = message.sender() {
+                        let classes = vec![
+                            "sender-text-red",
+                            "sender-text-orange",
+                            "sender-text-violet",
+                            "sender-text-green",
+                            "sender-text-cyan",
+                            "sender-text-blue",
+                            "sender-text-pink",
+                        ];
+
+                        let color_class = classes[user.id() as usize % classes.len()];
+                        imp.sender_label.add_css_class(color_class);
+                        if !self.imp().is_outgoing.get() {
+                            imp.separator
+                                .add_css_class(color_class.replace("text", "background").as_str());
+                        }
+
+                        *sender_color_class = Some(color_class.into());
+                    }
+                }
+                imp.sender_label.set_visible(true);
+            }
+
+            // Set content label expression
+            imp.message_label
+                .set_label(&strings::message_content(message));
+
+            let texture_binding = gtk::ConstantExpression::new(message)
+                .chain_closure::<Option<gdk::Texture>>(closure!(|_: Message, message: Message| {
+                    message_thumbnail_texture(message)
+                }))
+                .bind(&*imp.message_thumbnail, "paintable", Some(message));
+            bindings.push(texture_binding);
+        } else {
+            self.set_visible(false);
+        }
+
+        self.imp().is_loading.set(false);
+    }
+}
+
+fn message_thumbnail_texture(message: Message) -> Option<gdk::Texture> {
+    match message.content().0 {
+        MessageContent::MessageAnimation(data) => data.animation.minithumbnail,
+        MessageContent::MessageAudio(data) => data.audio.album_cover_minithumbnail,
+        MessageContent::MessageChatChangePhoto(data) => data.photo.minithumbnail,
+        MessageContent::MessageDocument(data) => data.document.minithumbnail,
+        MessageContent::MessagePhoto(data) => data.photo.minithumbnail,
+        MessageContent::MessageVideo(data) => data.video.minithumbnail,
+        _ => None,
+    }
+    .map(|thumbnail| {
+        gdk::Texture::from_bytes(&glib::Bytes::from_owned(glib::base64_decode(
+            &thumbnail.data,
+        )))
+        .unwrap()
+    })
+}
```

Comment:
I think you copied this from `SidebarRow`. Instead of repeating the code I think we should try to share it somehow. Maybe we can add a method to `BoxedMessageContent`:

```rs
impl BoxedMessageContent {
    fn minithumbnail(&self) -> tdlib::types::Minithumbnail {
        ...
    }
}
```